packages/smart-contracts/src/contracts/ERC20TransferableReceivable.sol

```diff
@@ -158,15 +158,16 @@ contract ERC20TransferableReceivable is ERC721, ERC721Enumerable, ERC721URIStora
    * @param paymentReference A reference for the payment.
    * @param amount The amount of ERC20 tokens to be paid.
    * @param erc20Addr The address of the ERC20 token to be used as payment.
-   * @param newTokenURI The URI to be set on the minted receivable token.
+   * @param requestID The ID of the request associated with the receivable.
+   *                  Useful for retrieving details of the request from the Request Network.
    * @dev Anyone can pay for the mint of a receivable on behalf of a user
    */
   function mint(
     address owner,
     bytes calldata paymentReference,
```

Comment:
Change Request: 

I'm struggling to understand why we need both `requestId` and `paymentReference` parameters when minting. I suspect we can save gas if we eliminated the `paymentReference` and used the `requestId` instead.

Reply:
The only problem I see is that requestId-based detection would be inconsistent with all other payment extensions, they all use `paymentReference` instead. Maybe @vrolland can detail the scenario when knowing one's requestId is a security weakness?

Reply:
This is a good point, I can eliminate paymentReference here.

Reply:
In the end, we decided to remove the `requestId` and keep the `paymentReference` - for privacy and consistency with other payment proxy contracts.